src/reminder/reminder.service.ts

```diff
@@ -0,0 +1,236 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Reminder } from 'src/entities/reminder.entity';
+import { Recurring } from 'src/entities/recurring.entity';
+import { Between, LessThan, MoreThanOrEqual, Repository } from 'typeorm';
+import { ModifiedReminder, GetReminder, ListReminderEachDate, ListUnfinishedReminder } from './reminder.interface';
+
+@Injectable()
+export class ReminderService {
+    constructor(
+        @InjectRepository(Reminder)
+        private reminderRepository: Repository<Reminder>,
+        @InjectRepository(Recurring)
+        private recurringRepository: Repository<Recurring>
+    ) { }
+
+    async deleteReminder(rid: number): Promise<string> {
+        const reminder = await this.reminderRepository.findOne({ where: { rid: rid } })
+        if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND)
+
+        await this.reminderRepository.delete(reminder);
+        return 'Complete'
+    }
+
+    async getReminder(rid: number): Promise<GetReminder> {
+        const reminder = await this.reminderRepository.findOne({ where: { rid: rid } })
+        if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND)
+
+        return reminder
+    }
+
+    async getFinishedReminder(currentDate: Date, uid: number): Promise<ListReminderEachDate[]> {
+        currentDate.setDate(currentDate.getDate() - 7)
+        currentDate.setHours(0, 0, 0)
+        const reminders: Reminder[] = await this.reminderRepository.find({
+            where: {
+                uid: uid,
+                isDone: true,
+                startingDateTime: MoreThanOrEqual(currentDate)
+            },
+            order: {
+                startingDateTime: "DESC"
+            }
+        })
+
+        currentDate.setDate(currentDate.getDate() + 8)
+        const listReminderGroupByDate: ListReminderEachDate[] = []
+        for (let i = 0; i < 9; i++) {
+            const listReminderEachDate: ModifiedReminder[] = []
+            for (const reminder of reminders) {
+                if (reminder.startingDateTime.getDate() === currentDate.getDate()) {
+                    listReminderEachDate.push({
+                        rid: reminder.rid,
+                        title: reminder.title,
+                        note: reminder.note,
+                        isRemindCaretaker: reminder.isRemindCaretaker,
+                        importanceLevel: reminder.importanceLevel,
+                        imageid: reminder.imageid,
+                        hour: reminder.startingDateTime.getHours(),
+                        minute: reminder.startingDateTime.getMinutes()
+                    })
+                    reminders.splice(reminders.indexOf(reminder), 1) //don't know if this gonna increase the speed or decrease
```

Comment:
If i'm not wrong, this line will delete the element you just pushed?

Reply:
เยป เพราะว่าจะได้ลบอนัที่แอดเข้าarrayไปแล้ว ละก็เผื่อมันจะช่วยเพิ่มperformanceด้วย แต่ถึงลบก็ไม่ส่งผลอะไร